src/babylon.scene.ts

```diff
@@ -1785,14 +1785,12 @@ module BABYLON {
 
             let type = PointerEventTypes.POINTERUP;
             if (this.onPointerObservable.hasObservers()) {
-                if (!clickInfo.ignore) {
-                    if (!clickInfo.hasSwiped) {
-                        if (clickInfo.singleClick && this.onPointerObservable.hasSpecificMask(PointerEventTypes.POINTERTAP)) {
-                            type = PointerEventTypes.POINTERTAP;
-                        }
-                        if (clickInfo.doubleClick && this.onPointerObservable.hasSpecificMask(PointerEventTypes.POINTERDOUBLETAP)) {
-                            type = PointerEventTypes.POINTERDOUBLETAP;
-                        }
+                if (!clickInfo.ignore && !clickInfo.hasSwiped) {
+                    if (clickInfo.singleClick && this.onPointerObservable.hasSpecificMask(PointerEventTypes.POINTERTAP)) {
+                        type = PointerEventTypes.POINTERTAP;
+                    }
+                    else if (clickInfo.doubleClick && this.onPointerObservable.hasSpecificMask(PointerEventTypes.POINTERDOUBLETAP)) {
+                        type = PointerEventTypes.POINTERDOUBLETAP;
                     }
                 }

```

Comment:
Like this you mean?

Sorry for the last minute restructure aswell..
just notised i forgot the bundled if's 

    if( x ){
      if( y ) {
        // something
      }
    }

    // instead of
    if( x && y ){
      // something
    }

Reply:
This looks good to me. I will do.orr testing and report back here